content/lessons/testing/component-testing-rtl.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Component Testing with React Testing Library
+template: lesson
+draft: false
+slug: /courses/Testing/component-testing-rtl
+course: Testing
+defaultTab: tests
+tags:
+  - Jest
+  - React Testing Library
+description: "This lesson covers testing React components, using Jest and React Testing
+Library to run tests, and how to make assertions on mounted React elements. We'll go into
+the difference between React Components and Elements, understand the purpose of the React
+Testing Library, and mount React elements in order to make assertions about them."
+timeToCompletion: ~1 - 2.5 hours
+videoLinks:
+  - https://www.youtube.com/embed/BUXJIg2Wve4?start=339
+preReadQuizLink: TODO
+readingLinks:
+  - link: https://reactjs.org/blog/2015/12/18/react-components-elements-and-instances.html
+    description: This great blog post by Dan Abramov covers the difference between React components, elements, and component instances.
+    title: Components, elements, and instances
+  - link: https://testing-library.com/docs/react-testing-library/intro
+    description: We will be using the React Testing Library docs throughout the session, so it would be helpful to get familiar with this resource.
+    title: React Testing Library Docs
+  - link: https://testing-library.com/docs/react-testing-library/cheatsheet
+    description: React Testing Library also provides a cheatsheet for quick reference to queries and events. This will come in handy during the lesson!
+    title: React Testing Library Cheatsheet
+  - link: https://kentcdodds.com/blog/introducing-the-react-testing-library
+    description: The blog post introducing React Testing Library to the world.
+    title: Introducing React Testing Library
+preReadQuiz:
+  description: Take this quiz to test your understanding of React Testing Library!
+  questions:
+    - description: Which of these should be used for testing asynchronous behavior?
+      choices:
+      - value: getByText()
+      - value: queryAllByRole()
+      - value: findByDisplayValue()
+      - value: queryByLabelText()
+      - value: findAllByAltText()
+      correctChoices:
+      - 2
+      - 4
+      explanation: When testing async behavior, always use the findBy* queries, which return a promise that resolves to the value of the element or elements you're querying for.
+      type: checkbox
+    - description: "True or false: If you want to inspect an element in the console, console.log is the best method."
```

Comment:
This question should clarify that the answer to this is false _as it specifically relates to React Testing Library_. There may be other uses for using `console.log` in a test (to debug data being passed into a component.)